File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
import sys
input = sys.stdin.readline

dwarf_list = [int(input()) for _ in range(9)]
dwarf_list.sort()
total_height = 0
a=0
b = 0

total_dwarf_tall = sum(dwarf_list)

for i in dwarf_list:
    for j in dwarf_list:
        if i != j:
            if total_dwarf_tall - i - j == 100:
                a = i
                b = j

dwarf_list.remove(a)
dwarf_list.remove(b)

print('\n'.join(map(str,dwarf_list)))

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.js
const input = require('fs')
    .readFileSync('/dev/stdin', 'utf-8')
    .trim()
    .split('\n')
    .map((item) => Number(item));

const totalHeight = input
    .map((item) => Number(item))
    .reduce((acc, cur) => {
        return acc + cur;
    }, 0);

const findOddKids = (arr, target) => {
    for (let i = 0; i < arr.length; i++) {
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[i] + arr[j] === target) {
                return [arr[i], arr[j]];
            }
        }
    }
};

const [kids1, kids2] = findOddKids(input, totalHeight - 100);
console.log(
    input
        .filter((height) => height !== kids1 && height !== kids2)
        .sort((a, b) => a - b)
        .join('\n')
);
